### PR TITLE
refactor(client): decompose logic from register modal form

### PR DESCRIPTION
**Pull request by gordonpn**
#

## Changes this PR introduces

- An example of how custom hooks can be used to abstract the logic away from functional components (to keep components easier to read)

## How to test this PR

- The functionality of the register modal should remain the same.

## ClickUp reference

CU-q37k4y

## Notes for reviewers

- Feel free to ask me anything

## Checklist

- [x] Have I renamed the title of this pull request to follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)?
- [x] I will squash and merge when my PR is approved.
- [x] I have added tests to cover my changes (if applicable).

### CODE REVIEW

**Comment by Mackbellemore** [comment]
Task linked: [CU-q37kzj Refactor ExportFiles to use a custom hook](https://app.clickup.com/t/q37kzj)

**Review by DavidLiang01 — APPROVED** [review]

File: client/src/hooks/useRegisterForm.jsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    } else {
+      setPasswordIsValidated(false);
+      setButtonIsDisabled(false);
+    }
+  };
+
+  // Validates email
+  const registerHandleEmailValidation = (e) => {
+    const email = e.target.value;
+    const regex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```
Comment (Mackbellemore) [inline client/src/hooks/useRegisterForm.jsx:68]: this is so painful to look at

Comment (gordonpn) [reply client/src/hooks/useRegisterForm.jsx:68]: you've seen it before

**Review by Mackbellemore — APPROVED** [review]

File: client/src/hooks/useRegisterForm.jsx (hunk 1 above)
Comment (Mackbellemore) [reply client/src/hooks/useRegisterForm.jsx:68]: i know just a reminder

Comment (Mackbellemore) [reply client/src/hooks/useRegisterForm.jsx:68]: never forget Mr. Pham

**Pull request merged**